### DISCUSSION TITLE
Ctrl + Enter does not execute the query on many terminals

### DISCUSSION THREAD

**Discussion by carteakey** [tconbeer/harlequin#598] [Q&A, Troubleshooting, Support] [answered]
Hey @tconbeer, very cool and useful project.

Running this is a MacOS default terminal, all commands work except Run Query. 

I tried all combinations (cmd + enter, ctrl + enter), but it results in a new line.

Any thoughts on what could be the issue?

**Comment by tconbeer** [comment]
**NOTE to those with this issue:**
- `ctrl + j` and `cmd + j` work to submit your query
- You can also click the (blue) footer, where it says `CTRL + ENTER: RUN QUERY`

I'll be improving this experience soon.

--- Original message ---
As a workaround, could you try `ctrl+j` and `cmd+j`?

I don't have access to a Mac -- if you could do me a favor, could you install the textual dev tools and run the keys app?

```bash
$ pip install textual[dev]
$ textual keys
```

Then while running that app, can you press `ctrl+enter` and `cmd+enter` and paste the output into this issue?

**Comment by carteakey** [comment]
Thanks for looking into it! `ctrl+j` works just fine, which is good enough I guess. 

Strangely enough, both `ctrl+enter` and `cmd+enter` show the below output in textual.

![image](https://github.com/tconbeer/harlequin/assets/40712659/e3b116b2-2585-48d7-a6b6-ae189287e6c5)

Let me know if I can help with any additional info.

**Comment by tconbeer** [comment]
Amazing. I'll update the docs for ctrl+j. (Different terminals support different key combinations)

**Comment by ned2** [comment]
I'm seeing this same issue in Ubuntu. Control+Enter does not work, but Control+j does.

I've tried this with both the default Ubuntu terminal as well as my primary terminal, Guake.

Edit: this appears this is a limitation of the underlying terminal emulators. when I run `showkeys -a` I see that both `enter` and `control`+`enter` sends the same key code.

from googling around, control+enter just seems to be something that doesn't really work across terminal emulators. the only workarounds I've seen involve remapping the codes sent by keyboard keys for the terminal emulator you're using. which I'm not really keen to dive into

**Comment by tconbeer** [comment]
Other than making ctrl+j better documented, is there a key combination either of you would think is more intuitive for submit? (I could also add a more obvious button, although the footer doubles as a row of buttons)

**Comment by ned2** [comment]
I reckon adding a submit button would be good regardless of whether another key combination is added. Would make it generally more accessible and folks would at least have that option if they've missed the documentation.

As for other combinations, how about control+space? that looks to have a different key code than space on its own in my terminal emulator.

**Comment by carteakey** [comment]
+1 for a button, since the UI already supports mouse clicks.

**Comment by hannes** [comment]
Same issue here, ctrl-j works however

**Comment by kbd** [comment]
I'm using kitty terminal and having this issue. Did y'all figure out *why* ctrl+enter doesn't work? It would be nice to be able to use ctrl+enter to execute queries. I've tried keymapping option+enter as well, unsuccessfully.

**Reply by tconbeer** [reply, marked as answer]
`ctrl+enter` is intercepted by your terminal and forwarded to your shell as a different virtual key, probably the plain `enter`. You need to change your terminal config to make it forward `ctrl+enter` as either `ctrl+enter` or `ctrl+j` (as windows terminal and others do), so that Harlequin (via your shell) receives the correct keypress. A little more info here: https://harlequin.sh/docs/troubleshooting/key-bindings
